packages/api/internal/template-manager/create_template.go

```diff
@@ -64,52 +54,12 @@ func (tm *TemplateManager) CreateTemplate(
 			StartCommand:       startCommand,
 		},
 	})
-	err = utils.UnwrapGRPCError(err)
-	if err != nil {
-		return fmt.Errorf("failed to create template '%s': %w", templateID, err)
-	}
 
-	// Wait for the build to finish and save logs
-	for {
-		_, receiveErr := logs.Recv()
-		if receiveErr == io.EOF {
-			break
-		} else if receiveErr != nil {
-			// There was an error during the build
-			return fmt.Errorf("error when building env: %w", receiveErr)
-		}
-	}
-
-	trailer := logs.Trailer()
-	rootfsSizeStr, ok := trailer[storage.RootfsSizeKey]
-	if !ok {
-		return fmt.Errorf("rootfs size not found in trailer")
-	}
-
-	diskSize, parseErr := strconv.ParseInt(rootfsSizeStr[0], 10, 64)
-	if parseErr != nil {
-		return fmt.Errorf("error when parsing rootfs size: %w", parseErr)
-	}
-
-	envdVersion, ok := trailer[storage.EnvdVersionKey]
-	if !ok {
-		return fmt.Errorf("envd version not found in trailer")
-	}
-
-	err = db.FinishEnvBuild(childCtx, templateID, buildID, diskSize, envdVersion[0])
-	if err != nil {
-		return fmt.Errorf("error when finishing build: %w", err)
-	}
-
-	telemetry.ReportEvent(childCtx, "created new environment", attribute.String("env.id", templateID))
-
-	cacheErr := buildCache.SetDone(templateID, buildID, api.TemplateBuildStatusReady)
-	if cacheErr != nil {
-		err = fmt.Errorf("error when setting build done in logs: %w", cacheErr)
-		telemetry.ReportCriticalError(childCtx, cacheErr)
+	if utils.UnwrapGRPCError(err) != nil {
+		return fmt.Errorf("failed to create template '%s': %w", templateID, err)
```

Comment:
we are using the unwrapped error at the other places, is there a reason why not to used it here